components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/handlers/UsernameValidationConfigurationHandler.java

```diff
@@ -0,0 +1,162 @@
+                configuration.setRules(rules);
+            }
+            return configuration;
+        } catch (InputValidationMgtException e) {
+            throw new InputValidationMgtException(ERROR_GETTING_EXISTING_CONFIGURATIONS.getCode(), e.getMessage());
+        }
+    }
+
+    @Override
+    public boolean validateValidationConfiguration(List<RulesConfiguration> configurationList)
+            throws InputValidationMgtClientException {
+
+        List<String> validatorNames = new ArrayList<>();
+        configurationList.forEach(config -> validatorNames.add(config.getValidatorName()));
+        int validConfigurations = 0;
+        for (RulesConfiguration configuration: configurationList) {
+            if (EmailFormatValidator.class.getSimpleName().equals(configuration.getValidatorName())) {
+                if (Boolean.parseBoolean(configuration.getProperties().get(ENABLE_VALIDATOR))) {
+                    validConfigurations += 1;
+                }
+                validatorNames.remove(EmailFormatValidator.class.getSimpleName());
+            } else if (AlphanumericValidator.class.getSimpleName().equals(configuration.getValidatorName())) {
+                if (Boolean.parseBoolean(configuration.getProperties().get(ENABLE_VALIDATOR))) {
+                    if (validatorNames.contains(LengthValidator.class.getSimpleName())) {
+                        validConfigurations += 1;
+                        validatorNames.remove(LengthValidator.class.getSimpleName());
+                    }
+                }
+                validatorNames.remove(AlphanumericValidator.class.getSimpleName());
+            }
+        }
+        if (validConfigurations != 1 || !validatorNames.isEmpty()) {
+            throw new InputValidationMgtClientException(ERROR_INVALID_VALIDATORS_COMBINATION.getCode(),
+                    String.format(ERROR_INVALID_VALIDATORS_COMBINATION.getDescription(), USERNAME));
+        }
+        return true;
+    }
+
+    private Map<String, String> getUsernameRegEx(RealmConfiguration realmConfig) {
+
+        Map<String, String> usernameRegEx = new HashMap<>();
+        if (MultitenantUtils.isEmailUserName()) {
+
+            if (StringUtils.isNotBlank(realmConfig.getUserStoreProperty(PROPERTY_USER_NAME_WITH_EMAIL_JS_REG_EX))) {
+
+                usernameRegEx.put(JAVA_REG_EX, realmConfig.getUserStoreProperty(
+                        PROPERTY_USER_NAME_WITH_EMAIL_JS_REG_EX).replaceAll("//", "/"));
+                usernameRegEx.put(JS_REG_EX, realmConfig.getUserStoreProperty(PROPERTY_USER_NAME_WITH_EMAIL_JS_REG_EX));
+                return usernameRegEx;
+            }
+
+            if ((StringUtils.isBlank(realmConfig.getUserStoreProperty(PROPERTY_USER_NAME_JAVA_REG_EX))
+                    || StringUtils.isBlank(realmConfig.getUserStoreProperty(PROPERTY_USER_NAME_JS_REG_EX)))
+                    && (StringUtils.isBlank(realmConfig.getUserStoreProperty(PROPERTY_USER_NAME_JAVA_REG_EX))
```

Comment:
The same condition is duplicated here. PROPERTY_USER_NAME_JAVA_REG_EX > PROPERTY_USER_NAME_JAVA_REG

Reply:
Addressed